packages/tizen_bundle/lib/tizen_bundle.dart

```diff
@@ -0,0 +1,347 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:tizen_interop/4.0/tizen.dart';
+import 'package:tizen_log/tizen_log.dart';
+
+/// The class for the property type of the bundle.
+class _BundleTypeProperty {
+  /// The array type.
+  static const int array = 0x0100;
+
+  /// The primitive type.
+  static const int primitive = 0x0200;
+
+  /// The measurable type.
+  static const int measurable = 0x0400;
+}
+
+/// The class for the type of the bundle.
+class BundleType {
+  /// The none type.
+  static const int none = 0;
+
+  /// The string type.
+  static const int string = 1 | _BundleTypeProperty.measurable;
+
+  /// The string list Type.
+  static const int strings = string | _BundleTypeProperty.array;
+
+  /// The byte type.
+  static const int byte = 2;
+
+  /// The byte list type.
+  static const int bytes = byte | _BundleTypeProperty.array;
+}
+
+/// The class for information of the key of the bundle object.
+class KeyInfo {
+  /// The name of the key.
+  final String name;
+
+  /// The type of the key.
+  final int type;
+
+  /// Creates an instance of [KeyInfo] with the given arguments.
+  KeyInfo(this.name, this.type);
+
+  /// Check whether the type of the key is array or not.
+  bool get isArray => (type & _BundleTypeProperty.array) as bool;
+}
+
+class _BundleErrorFactory {
+  static final _instance = _BundleErrorFactory._internal();
+  final Map<int, String> _errorMessages = {
+    bundle_error_e.BUNDLE_ERROR_OUT_OF_MEMORY: 'Out of memory',
+    bundle_error_e.BUNDLE_ERROR_INVALID_PARAMETER: 'Invalid parameter',
+    bundle_error_e.BUNDLE_ERROR_KEY_EXISTS: 'Key already exists',
+    bundle_error_e.BUNDLE_ERROR_KEY_NOT_AVAILABLE: 'Key does not exists',
+  };
+
+  factory _BundleErrorFactory() {
+    return _instance;
+  }
+
+  _BundleErrorFactory._internal();
+
+  void throwException(int error) {
+    if (_errorMessages.containsKey(error)) {
+      throw Exception(_errorMessages[error]);
+    } else {
+      throw Exception('Unknown error: $error');
+    }
+  }
+}
+
+/// A bundle raw object represents a bundle_raw.
+/// A bundle raw
+class BundleRaw {
+  /// The raw data.
+  final String raw;
+
+  /// The length of the raw data.
+  final int length;
```

Comment:
Is it different with `raw.length`?

Reply:
It's same. I've changed the implementation. The return type is changed to String. And then, the BundleRaw class is removed.